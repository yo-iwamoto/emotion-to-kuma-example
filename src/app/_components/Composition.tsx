'use client';

import { css } from '@emotion/react';

export function Composition() {
  return <p css={[styles.text, styles.bold]}>Composition</p>;
}

const colorStyle = css`
  color: skyblue;
`;

const FONT_STYLES = {
  lg: 40,
};

const styles = {
  text: css`
    ${colorStyle}
    font-size: ${FONT_STYLES.lg}px;
  `,
  bold: css`
    font-weight: bold;
  `,
};
