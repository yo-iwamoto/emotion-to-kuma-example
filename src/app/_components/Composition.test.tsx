import { Composition } from './Composition';
import { render } from '@testing-library/react';

describe('Composition', () => {
  it('renders unchanged', () => {
    const { asFragment } = render(<Composition />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('has color style', () => {
    const { getByText } = render(<Composition />);

    expect(getByText('Composition')).toHaveStyleRule('color', 'skyblue');
  });
});
