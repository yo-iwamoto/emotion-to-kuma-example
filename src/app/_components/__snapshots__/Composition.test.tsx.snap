// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Composition > renders unchanged 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: skyblue;
  font-size: 40px;
  font-weight: bold;
}

<p
    class="emotion-0"
  >
    Composition
  </p>
</DocumentFragment>
`;
